drivers/watchdog/wdt_cc32xx.c

```diff
@@ -0,0 +1,177 @@
+#include <inc/hw_wdt.h>
+#include <driverlib/pin.h>
+#include <driverlib/rom.h>
+#include <driverlib/rom_map.h>
+#include <driverlib/prcm.h>
+#include <driverlib/wdt.h>
+
+#define MAX_RELOAD_VALUE        0xFFFFFFFF
+
+#define LOG_LEVEL CONFIG_WDT_LOG_LEVEL
+#include <logging/log.h>
+#include <logging/log_ctrl.h>
+LOG_MODULE_REGISTER(wdt_cc32xx);
+
+struct wdt_cc32xx_data {
+	const unsigned long reg;
+	int reload;
+	wdt_callback_t cb;
+	uint8_t flags;
+};
+
+static uint32_t wdt_cc32xx_msToTicks(uint32_t ms)
+{
+	static const uint32_t ratio = CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / 1000;
+	static const uint32_t maxMs = MAX_RELOAD_VALUE / ratio;
+
+	if (ms > maxMs) {
+		return maxMs;
+	}
+
+	return ms * ratio;
+}
+
+static int wdt_cc32xx_enable(const struct device *dev)
+{
+	struct wdt_cc32xx_data *data = dev->data;
+	const uint32_t reload = wdt_cc32xx_msToTicks(data->reload);
+
+	MAP_WatchdogIntClear(data->reg);
+	MAP_WatchdogReloadSet(data->reg, reload);
+	MAP_WatchdogEnable(data->reg);
+	LOG_DBG("Enabled");
+	return 0;
+}
+
+static int wdt_cc32xx_setup(const struct device *dev, uint8_t options)
+{
+	struct wdt_cc32xx_data *data = dev->data;
+	int rv;
+
+	if (options & WDT_OPT_PAUSE_IN_SLEEP) {
+		return -ENOTSUP;
+	}
+
+	MAP_WatchdogUnlock(data->reg);
+	if (options & WDT_OPT_PAUSE_HALTED_BY_DBG) {
+		MAP_WatchdogStallEnable(data->reg);
+	} else {
+		MAP_WatchdogStallDisable(data->reg);
+	}
+	rv = wdt_cc32xx_enable(dev);
+	MAP_WatchdogLock(data->reg);
+	return rv;
+}
+
+static int wdt_cc32xx_disable(const struct device *dev)
+{
+	return -ENOTSUP;
+}
+
+static int wdt_cc32xx_install_timeout(const struct device *dev,
+				      const struct wdt_timeout_cfg *cfg)
+{
+	struct wdt_cc32xx_data *data = dev->data;
+
+	if (COND_CODE_1(CONFIG_WDT_MULTISTAGE, (cfg->next), (0))) {
+		return -ENOTSUP;
+	}
+
+	data->reload = cfg->window.max;
+	data->cb = cfg->callback;
+	data->flags = cfg->flags;
+	LOG_DBG("Reload time %d", data->reload);
+	return 0;
+}
+
+static int wdt_cc32xx_feed(const struct device *dev, int channel_id)
+{
+	struct wdt_cc32xx_data *data = dev->data;
+	const uint32_t reload = wdt_cc32xx_msToTicks(data->reload);
+	bool inIsr = k_is_in_isr();
+
+	if (!inIsr) {
+		MAP_WatchdogUnlock(data->reg);
+	}
+	MAP_WatchdogIntClear(data->reg);
+	MAP_WatchdogReloadSet(data->reg, reload);
+	if (!inIsr) {
+		MAP_WatchdogLock(data->reg);
+	}
+	LOG_DBG("Feed");
+	return 0;
+}
+
+static void wdt_cc32xx_isr(const struct device *dev)
+{
+	struct wdt_cc32xx_data *data = dev->data;
+
+	LOG_DBG("ISR");
+	if (data->cb) {
+		data->cb(dev, 0);
+	}
+	if (data->flags != WDT_FLAG_RESET_NONE) {
+		LOG_PANIC();
+		MAP_PRCMMCUReset(data->flags & WDT_FLAG_RESET_SOC);
+		while (1) {
+		}
+	}
+}
+
+static int wdt_cc32xx_init(const struct device *dev)
+{
+	struct wdt_cc32xx_data *data = dev->data;
+	int rv;
+
+	LOG_DBG("init");
+	IRQ_CONNECT(DT_INST_IRQN(0),
+		    DT_INST_IRQ(0, priority),
+		    wdt_cc32xx_isr, DEVICE_DT_INST_GET(0), 0);
+	irq_enable(DT_INST_IRQN(0));
```

Comment:
Are there any SoCs with multiple instances of the watchdog hardware?

Reply:
For cc32xx family only one watchdog instance present. Do you want me to use `DT_INST_FOREACH_STATUS_OKAY` for consistency?